tests/test_runs/test_run_functions.py

```diff
@@ -518,6 +518,18 @@ def _run_and_upload_regression(
             sentinel=sentinel,
         )
 
+    def test_upload_run_with_setup_string(self):
+        SETUP_STRING = "setup-string"
+        run = openml.runs.run_model_on_task(
+            DummyClassifier(strategy="prior"),
+            self.TEST_SERVER_TASK_SIMPLE[0],
+            avoid_duplicate_runs=False,
+        )
+        run.setup_string = SETUP_STRING
+        run.publish()
```

Comment:
Do we need the server round trip here to assert that this is working or would an xml round trip suffice?

Reply:
What are you proposing here? That we check the run xml and don't actually publish the run? I'd considered it and prefer that actually, but this decoupling is not generally done in our unit tests which is why I did it this way.
We can also forgo doing an actual run, have something like:

```
run = OpenMLRun(task_id=0, flow_id=0, dataset_id=0, setup_string=SETUP_STRING)
xml = run._to_xml()
run = xmltodict.parse(xml)["oml:run"]
assert "oml:setup_string" in run
assert run["oml:setup_string"] == SETUP_STRING
```

should make the test take virtually no time, and also avoid disk operations.

Reply:
> That we check the run xml and don't actually publish the run?

Yes, I was thinking about that and was wondering if it's possible.

> but this decoupling is not generally done in our unit tests

Yeah, it would be great to have that, but I guess we can't easily get that :+1: 

>  We can also forgo doing an actual run, have something like:

Hm, maybe we should initialize the Run() ?

Reply:
Not sure if it would work, though: `openml.runs.function._create_run_from_xml(xml, from_server=False)`

Reply:
> Hm, maybe we should initialize the Run() ?

Doesn't the first line take care of that?

Reply:
Ah, didn't see the follow up message.

Reply:
Updated (and test moved to other file).